IotcRestApi/regulator_draft.md

```diff
@@ -0,0 +1,145 @@
+# API
+Bude se jednat o REST API, bude k dispozici Swagger.
+
+# Zabezpečení
+Komunikace bude zabezpečena bearer tokenem v hlavičce, informace budou doplněny.
+
+Většina endpointů přijímajících data v těle requestu vyžaduje položku `requestId`. Jedná se o jednoznačný identifikátor requestu, který generuje partner (jedná se o GUID) a slouží pro zajištění idempotence, čímž je docíleno toho, že akce je provedena jen jednou pokud dojde o opakovanému zavolání v určitém časovém intervalu.
+
+# Chybové odpovědi
+Chyby 5xx jsou způsobené chybou serveru a neměli by nikdy nastat. Pokud server vratí tuto chybu tak
+by měl být okamžitě informován zástupce z firmy Netlia.
+
+Chyby 4xx jsou vráceny, pokud klient provedl neplatný/nevalidní request. Vysvětlení stavových kódu je možné najít
+v HTTP dokumentaci - https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses. Podporované stavové kódy budou doplněny.
+
+V některých případech 4xx error obsahuje body s dalšimy informacemi o chybě. Body má následující formát:
+
+```yaml
+{
+    "errors" : [
+        {
+            "errorCode":int
+            "description":string
+            "additionalInformation":string
+            "traceId":string
+        }
+    ]
+}
+```
+
+# Zařízení a podporované endpointy
+
+## Teplotní regulátor
+
+### POST api/temperature-regulator/{DeviceId}/mode
+
+Změna módu teplotního regulátoru.
+
+Předávané parametry:
+
+| Parametr    | Typ         | Povinný | Popis                               |
+|:------------|:------------|:--------|:------------------------------------|
+| requestId   | string-guid | ano     | Jednoznačný identifikátor requestu. |
+| mode        | string      | ano     | Cílový mód zařízení.                |
+
+Cílový mód zařízení může nabývat těchto hodnot:
+
+| Hodnota            | Název                       |
+|--------------------|-----------------------------|
+| basic-regulator    | Základní regulace teploty.  |
+| manual             | Manuální regulace.          |
+| summer             | Letní režim.                |
+
+Ukázka requestu:
```

Comment:
Mozna bych do te ukazky treba na zacatek dal i ukazku URL.

Reply:
bude k tomu swagger, tak je to zbytečné ne?

Reply:
To je zajimava otazka. Jestli to ma slouzit jen jako jednoducha ukazka jak bude API vypadat tak asi jo.

Rikam si ted ze az bude existovat ten swagger tak mi to tam muzeme zdokumentovat uplne cele a tenhle dokument vubec nemit.

Reply:
No to jsem si tak rikal ze by bylo nejlepsi - mit tohle jen odkaz na swagger a pripedne jen doplnujici info co by neslo dat tam. Protoze tam si to muze realne i vyzkouset a vidi hned ukazky requestu a response.